shared-scripts.sh

```diff
@@ -132,6 +132,72 @@ function _lint {
   loadCacheFromBuildJob
   chmod +x codebuild_specs/scripts/lint_pr.sh && ./codebuild_specs/scripts/lint_pr.sh
 }
+function _verifyAmplifyBackendCompatability {
+  echo "Verify Amplify Backend Compatibility"
+  loadCacheFromBuildJob
+
+  # Unset container credentials environment variables since some of the tests in packages/cli/src/command_middleware.test.ts 
+  # expect not to fetch the credentials. This is to avoid the tests from failing.
+  echo "Unsetting container credentials environment variables"
+  unset AWS_CONTAINER_CREDENTIALS_RELATIVE_URI
+  unset AWS_CONTAINER_CREDENTIALS_FULL_URI
+
+  # 1. Install NVM and set up
+  echo "Installing NVM and setting Node.js version"
+  curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/master/install.sh | bash
+  export NVM_DIR="$HOME/.nvm"
+  [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+  nvm install 18.20.4
+  nvm use 18.20.4
+  # Echo to verify the Node.js version
+  echo "Node.js version in use:"
+  node -v
+  # Increase buffer size to avoid error when git operations return large response on CI
```

Comment:
This is a good example of a comment: it explains why we're doing this thing that's not obvious from just reading the code itself. It could be improved by explaining why that particular magic number is the right one. That will help us understand why that value is in use, whether it's safe to modify later, etc.